components/clp-package-utils/clp_package_utils/general.py

```
@@ -42,15 +46,50 @@
-
 DOCKER_MOUNT_TYPE_STRINGS = ["bind"]
 
 
+class DockerDependencyError(OSError):
+    """Base class for errors related to Docker dependencies."""
+
+
+class DockerNotAvailableError(DockerDependencyError):
+    """Raised when Docker or Docker Compose is unavailable."""
+
+    def __init__(self, base_message: str, process_error: subprocess.CalledProcessError):
```

Ruff (0.14.1)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)